vochain/transaction.go

```diff
@@ -185,54 +185,139 @@ func (app *BaseApplication) AddTx(vtx *VochainTx, commit bool) (*AddTxResponse,
+					if txCost != 0 {
+						if err := app.State.BurnTxCost(faucetIssuerAddress, txCost); err != nil {
+							return nil, fmt.Errorf("setAccountTx: burnTxCost %w", err)
+						}
 					}
 					faucetPayload := &models.FaucetPayload{}
 					if err := proto.Unmarshal(tx.FaucetPackage.Payload, faucetPayload); err != nil {
-						return nil, fmt.Errorf("could not unmarshal faucet package: %w", err)
+						return nil, fmt.Errorf("createAccountTx: cannot unmarshal faucetPayload %w", err)
 					}
-					// consume provided faucet payload
-					return response, app.State.ConsumeFaucetPayload(
-						txValues.FaucetPayloadSigner,
-						&models.FaucetPayload{
-							Identifier: faucetPayload.GetIdentifier(),
-							To:         faucetPayload.GetTo(),
-							Amount:     faucetPayload.GetAmount(),
-						},
-						models.TxType_SET_ACCOUNT_INFO,
+					if err := app.State.ConsumeFaucetPayload(
+						faucetIssuerAddress,
+						faucetPayload,
+					); err != nil {
+						return nil, fmt.Errorf("setAccountTx: consumeFaucet %w", err)
+					}
+					// transfer balance from faucet package issuer to created account
+					return response, app.State.TransferBalance(
+						faucetIssuerAddress,
+						txSenderAddress,
+						faucetPayload.Amount,
+					)
+				}
+				return response, nil
+
+			case models.TxType_SET_ACCOUNT_INFO_URI:
+				txSenderAddress, err := ethereum.AddrFromSignature(vtx.SignedBody, vtx.Signature)
+				if err != nil {
+					return nil, fmt.Errorf("createAccountTx: txSenderAddress %w", err)
+				}
+				// consume cost for setAccount
+				if err := app.State.BurnTxCostIncrementNonce(
+					txSenderAddress,
+					models.TxType_SET_ACCOUNT_INFO_URI,
+				); err != nil {
+					return nil, fmt.Errorf("setAccountTx: burnCostIncrementNonce %w", err)
+				}
+				txAccount := common.BytesToAddress(tx.GetAccount())
+				if txAccount != (common.Address{}) {
+					return response, app.State.SetAccountInfoURI(
+						txAccount,
+						tx.GetInfoURI(),
 					)
 				}
-				// any faucet payload provided, just create account
-				return response, app.State.CreateAccount(txValues.TxSender,
+				return response, app.State.SetAccountInfoURI(
+					txSenderAddress,
 					tx.GetInfoURI(),
-					make([]common.Address, 0),
-					0,
 				)
+
+			case models.TxType_ADD_DELEGATE_FOR_ACCOUNT:
+				txSenderAddress, err := ethereum.AddrFromSignature(vtx.SignedBody, vtx.Signature)
+				if err != nil {
+					return nil, fmt.Errorf("createAccountTx: txSenderAddress %w", err)
+				}
+				if err := app.State.BurnTxCostIncrementNonce(
+					txSenderAddress,
+					models.TxType_ADD_DELEGATE_FOR_ACCOUNT,
+				); err != nil {
+					return nil, fmt.Errorf("setAccountDelegate: burnTxCostIncrementNonce %w", err)
+				}
+				if err := app.State.SetAccountDelegate(
+					txSenderAddress,
+					tx.Delegates,
+					models.TxType_ADD_DELEGATE_FOR_ACCOUNT,
+				); err != nil {
+					return nil, fmt.Errorf("setAccountDelegate: %w", err)
+				}
+			case models.TxType_DEL_DELEGATE_FOR_ACCOUNT:
```

Comment:
Nit: let's add either a blank line between "case" statements or not (I think we should add them here).

Reply:
I guess you are saying this because of the next switch case, I've added a new line on those because is less clear when reading because you have more code, but in the former I do not think is necessary cause someone can clearly distinguish between each switch case.
Said so, I've added it as you are recommending.